public/app/features/serviceaccounts/ServiceAccountTable.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useMemo } from 'react';
+import Skeleton from 'react-loading-skeleton';
+
+import {
+  Avatar,
+  CellProps,
+  Column,
+  InteractiveTable,
+  Pagination,
+  Stack,
+  TextLink,
+  Button,
+  IconButton,
+  Icon,
+} from '@grafana/ui';
+import { UserRolePicker } from 'app/core/components/RolePicker/UserRolePicker';
+import { contextSrv } from 'app/core/core';
+import { AccessControlAction, OrgRole, Role, ServiceAccountDTO } from 'app/types';
+
+import { OrgRolePicker } from '../admin/OrgRolePicker';
+
+type Cell<T extends keyof ServiceAccountDTO = keyof ServiceAccountDTO> = CellProps<
+  ServiceAccountDTO,
+  ServiceAccountDTO[T]
+>;
+
+interface ServiceAccountTableProps {
+  services: ServiceAccountDTO[];
+  onRoleChange: (role: OrgRole, serviceAccount: ServiceAccountDTO) => void;
+  roleOptions: Role[];
+  onRemoveButtonClick: (serviceAccount: ServiceAccountDTO) => void;
+  onDisable: (serviceAccount: ServiceAccountDTO) => void;
+  onEnable: (serviceAccount: ServiceAccountDTO) => void;
+  onAddTokenClick: (serviceAccount: ServiceAccountDTO) => void;
+  showPaging?: boolean;
+  totalPages: number;
+  onChangePage: (page: number) => void;
+  currentPage: number;
+  isLoading: boolean;
+}
+
+export const ServiceAccountTable = ({
+  services,
+  onRoleChange,
+  roleOptions,
+  onRemoveButtonClick,
+  onDisable,
+  onEnable,
+  onAddTokenClick,
+  showPaging,
+  totalPages,
+  onChangePage,
+  currentPage,
+  isLoading,
+}: ServiceAccountTableProps) => {
+  const columns: Array<Column<ServiceAccountDTO>> = useMemo(
+    () => [
+      {
+        id: 'avatarUrl',
+        header: '',
+        cell: ({ cell: { value }, row: { original } }: Cell<'role'>) => {
+          return getCellContent(value, original, isLoading, 'avatarUrl');
+        },
+        sortType: 'string',
```

Comment:
Do you actually need sort type here? I don't think it makes sense to have sorting enabled for this column

Reply:
Nope, I took a look at the original table and it does not sort so I think we could keep just sorting by 'account'. Wdyt?

Reply:
Sounds like an idea :)

Reply:
Done ;) LGTY?